tmail-backend/rate-limiter/rate-limiter-api/src/main/scala/com/linagora/tmail/rate/limiter/api/CacheRateLimitingPlan.scala

```diff
@@ -0,0 +1,96 @@
+
+  private val cacheLoaderGet: AsyncCacheLoader[RateLimitingPlanId, RateLimitingPlan] =
+    (key: RateLimitingPlanId, executor: Executor) => Mono.from(repository.get(key))
+      .subscribeOn(Schedulers.fromExecutor(executor))
+      .toFuture
+
+  private val cacheLoaderPlanExists: AsyncCacheLoader[RateLimitingPlanId, java.lang.Boolean] =
+    (planId: RateLimitingPlanId, executor: Executor) => Mono.from(repository.planExists(planId))
+      .subscribeOn(Schedulers.fromExecutor(executor))
+      .toFuture
+
+  private val cacheLoaderList: AsyncCacheLoader[String, util.List[RateLimitingPlan]] =
+    (_: String, executor: Executor) => Flux.from(repository.list())
+      .subscribeOn(Schedulers.fromExecutor(executor)).collectList()
+      .toFuture
+
+  private val gaugePrefixValue: String = gaugePrefix.map(_ + ".").getOrElse("")
+
+  private val cacheGet: AsyncLoadingCache[RateLimitingPlanId, RateLimitingPlan] = {
+    val loadingCache: AsyncLoadingCache[RateLimitingPlanId, RateLimitingPlan] = Caffeine.newBuilder()
+      .expireAfterWrite(expireDuration)
+      .recordStats()
+      .buildAsync[RateLimitingPlanId, RateLimitingPlan](cacheLoaderGet)
+
+    gaugeRegistry.register(gaugePrefixValue + "rate_limiting_plan.cache.get.hitRate", () => loadingCache.synchronous().stats().hitRate())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.get.missCount", () => loadingCache.synchronous().stats().missCount())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.get.hitCount", () => loadingCache.synchronous().stats().hitCount())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.get.size", () => loadingCache.synchronous().estimatedSize())
+    loadingCache
+  }
+
+  private val cachePlanExists: AsyncLoadingCache[RateLimitingPlanId, lang.Boolean] = {
+    val loadingCache: AsyncLoadingCache[RateLimitingPlanId, lang.Boolean] = Caffeine.newBuilder()
+      .expireAfterWrite(expireDuration)
+      .recordStats()
+      .buildAsync[RateLimitingPlanId, java.lang.Boolean](cacheLoaderPlanExists)
+
+    gaugeRegistry.register(gaugePrefixValue + "rate_limiting_plan.cache.planExists.hitRate", () => loadingCache.synchronous().stats().hitRate())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.planExists.missCount", () => loadingCache.synchronous().stats().missCount())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.planExists.hitCount", () => loadingCache.synchronous().stats().hitCount())
+      .register(gaugePrefixValue + "rate_limiting_plan.cache.planExists.size", () => loadingCache.synchronous().estimatedSize())
+    loadingCache
+  }
+
+  private val cacheList: AsyncLoadingCache[String, util.List[RateLimitingPlan]] = {
+    val loadingCache: AsyncLoadingCache[String, util.List[RateLimitingPlan]] = Caffeine.newBuilder()
+      .expireAfterWrite(expireDuration)
+      .recordStats()
+      .buildAsync[String, java.util.List[RateLimitingPlan]](cacheLoaderList)
```

Comment:
We should have only one chache IMO

Reply:
I think so, but after long consideration. IMO no way better

Reply:
- We have 3 methods that need cache (reads method): get, list, planExists. (Although Enforce mailet use only one method - `get`, But I think we should keep all)
- If we only cache List[RateLimitingPlan], and use it for all method, it will slower performance